src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2031,6 +2268,259 @@ object Desugar {
       for { exprs <- sequence(trigger.exps map exprD(ctx)) } yield in.Trigger(exprs)(src)
     }
 
+    def generateBuiltInFunction(tag: BuiltInFunctionTag, args: Vector[in.Type])(src: Meta): in.FunctionMember = {
```

Comment:
I did not check any of the generateBuiltIng methods.

Reply:
Maybe one comment. If I understood this correctly, then you are essentially adding the encoding of the buitlin operations at the desugarer. I am not entirely sure whether this is a good idea. For this PR it is fine, but in generel the idea is that the desugarer does not perform the encoding. Instead, I would just generate the methods, predicates, and functions in the typeencoding of channels.

Reply:
Does this imply that the internal representation needs a node that stores the tag and the specific types with which it is instantiated such that the encoding can handle specific tags?

Reply:
All function and method calls and predicate instances are translated to the usual internal AST node. Then you just have to make sure that the corresponding methods, functions, and predicates are generates. So, from my point of view, no additional nods is needed.

Reply:
I moved these `generate...` functions from the Desugarer to `BuiltInMembersImpl` located in the encoding part